examples/integration_with_error.py

```diff
@@ -45,61 +42,94 @@
 data /= 0.5 * 0.25  # the dt in the integral for both dims
 data /= sqrt(ndshape(data)["ph1"] * ndshape(data)["ph2"])  # normalization
 # }}}
-dt = diff(data.getaxis('t2')[r_[0,1]]).item()
-data.ft('t2',shift=True)
-data /= sqrt(ndshape(data)['t2']) * dt
-error_pathway = (set(((j,k) for j in range(ndshape(data)['ph1']) for k in range(ndshape(data)['ph2'])))
-        - set(excluded_pathways)
-        - set([(signal_pathway['ph1'],signal_pathway['ph2'])]))
-error_pathway = [{'ph1':j,'ph2':k} for j,k in error_pathway]
-s_int,frq_slice,mystd = integral_w_errors(data,signal_pathway,error_pathway,
-        indirect='vd', fl=fl,return_frq_slice=True)
-logger.debug(strm("check the std after FT", std(data["ph1", 0]["ph2", 0].data.real)))
-# the sqrt on the next line accounts for the var(real)+var(imag)
+# {{{ First, run the code that automatically chooses integration bounds
+# and also assigns error
 fl.next("compare manual vs. automatic", legend=True)
-# here I plot w/ manually chosen integration bounds:
-manual_bounds = data["ph1", 0]["ph2", 1]["t2":frq_slice]
-N = ndshape(manual_bounds)["t2"]
-df = diff(data.getaxis("t2")[r_[0, 1]]).item()
-logger.debug(
-    strm(ndshape(manual_bounds), "df is", df, "N is", N, "N*df is", N * df)
-)
-manual_bounds.integrate("t2")
-std_off_pathway = (
-    data["ph1", 0]["ph2", 0]["t2":bounds]
-    .C.run(lambda x: abs(x)**2/2)
-    .mean_all_but(["t2","vd"])
-    .mean("t2")
-    .run(sqrt)
+error_pathway = (
+    set(((j, k) for j in range(2) for k in range(4)))
+    - set(excluded_pathways)
+    - set([(signal_pathway["ph1"], signal_pathway["ph2"])])
 )
+error_pathway = [{"ph1": j, "ph2": k} for j, k in error_pathway]
+data = integral_w_errors(data, signal_pathway, error_pathway, fl=fl)
+fl.plot(data, ".", label="auto: real", capsize=6)
+fl.plot(data.imag, ".", label="auto: imaginary", capsize=6)
+# }}}
+logger.debug(strm("check the std after FT", std(data["ph1", 0]["ph2", 0].data.real)))
+# the sqrt on the next line accounts for the var(real)+var(imag)
+std_off_pathway = data["ph1", 0]["ph2", 0].C.run(lambda x: abs(x) ** 2).mean_all_but(
+    "t2"
+).mean("t2").run(sqrt) / sqrt(2)
+logger.debug(strm("check the std after FT -- manually", std_off_pathway))
```

Comment:
here I used the new order where there seemed to be a point, but otherwise left the old order in place